### PR TITLE
feat: remove role within an organization

### PR DESCRIPTION
**Pull request by coderoyalty**
## Description: What this PR is.

Provides an API endpoint removing a single role within an organization.

## How it can be tested.
Make a `DELETE` request to the endpoint `/api/v1/organisations/:org_id/roles/:role_id`.

## Checklist of What I did
- [x] Implement `DELETE /api/v1/organisation/{org-id}/roles/{role-id}` endpoint
- [x] Handle cases where the role or organization does not exist
- [x] Provide documentation for the endpoint
- [x] Provide unit test for the endpoint service

## Reference Issue

#256 

Issue on linear: https://linear.app/team-bingo/issue/BAC2-17/remove-role-within-an-organization-or-256

### CODE REVIEW

File: src/modules/organisation-role/organisation-role.controller.ts
Comment (Heba-WebDev) [inline src/modules/organisation-role/organisation-role.controller.ts:1]: We need a guard for this controller, so not any one can remove the role.

Comment (coderoyalty) [reply src/modules/organisation-role/organisation-role.controller.ts:1]: the ownership guard is globally available by the controller.

**Review by Heba-WebDev — APPROVED** [review]

File: src/modules/organisation-role/organisation-role.controller.ts
Comment (CFCIfe) [reply src/modules/organisation-role/organisation-role.controller.ts:1]: valid

File: src/modules/organisation-role/organisation-role.service.ts [hunk 1]
```diff
@@ -161,4 +162,51 @@ export class OrganisationRoleService {
...
+        message: 'Role successfully removed',
+      };
+    } catch (err) {
+      CustomExceptionHandler(err);
+
+      throw new InternalServerErrorException({
+        message: 'Something went wrong!',
+        status_code: 500,
+      });
+    }
```
Comment (CFCIfe) [inline src/modules/organisation-role/organisation-role.service.ts:210]: ResponseInterceptor handles this. remove try..catch

Comment (coderoyalty) [reply src/modules/organisation-role/organisation-role.service.ts:210]: done!

**Review by Heba-WebDev — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by CFCIfe — APPROVED** [review]

**Pull request merged**